src/lib/server/db/session.ts

```diff
@@ -96,6 +100,31 @@ export async function invalidateSession(sessionId: string): Promise<void> {
 	}
 }
 
+// Invalidate sessions that exceed the maximum number of sessions
+export async function invalidateExcessSessions(userId: number): Promise<number> {
+	// Assert that userId is valid
+	if (userId === null || userId === undefined) {
+		throw new Error('Invalid userId');
+	}
+
+	const newestSessions = await db
+		.select({
+			id: sessionTable.id
+		})
+		.from(sessionTable)
+		.where(eq(sessionTable.userId, userId))
+		.orderBy(desc(sessionTable.timestamp))
+		.limit(MAX_ACTIVE_SESSIONS);
+
+	const sessionIdsToKeep = newestSessions.map((session) => session.id);
+
+	const deletedSessions = await db
```

Comment:
w/o `returning()` the deletedSessions is of type "never[]", but also why would we even want to know this info? I would just change this function to return void and delete this const deletedSessions